### DISCUSSION TITLE
Number of dependencies are misconceived.

### DISCUSSION THREAD

**Discussion by jimmywarting** [npm/feedback#541] [Website] [answered]
One decision making in picking any dependency is also the number of sub dependencies it has. npmjs.org don't mention this in any meaningful way. 

We all want the stack to be small, fast to download, exec fast, less JIT compiling and not have so many places where vulnerability can enter. 
this is not taken into consideration when a developer looks at a npm package and thinking "grate, this only has one dependency" while in fact it has many sub-dependencies

taking [standard-markdown](https://npmgraph.js.org/?q=standard-markdown@7.0.0) as an example...

This is what i would like to see:
![image](https://user-images.githubusercontent.com/1148376/131309719-d03c8bfe-f563-47f5-9159-a24f4fe4115b.png)

This makes you 2nd guessing if you really need "that" package cuz it went from 6 to 254 dependencies

a very large/deep dependency can take long time to load if you take out the bundler/npm & tree shaking out of the equation.
this [playground](https://jspm.org/sandbox#H4sIAAAAAAAAA8VaW3PjthV+31+BqC9JuiCtiy/r2B5tk5207rqTWY+n00eIhCTYIMEAoGy5k//eA5CUKIo8lBzPdPZik/j44eDg4FwAXH0Xq8iuM06WNpE3H66qH5zFNx8IuUq4ZSRaMm24vR7kdk4vBr7BCiv5zUPqf8ZXYfG8+SRlCb8erAR/zpS2AxKp1PIUKJ5FbJfXMV+JiFP/8JGIVFjBJDURk/x6CB1chYUEVzMVrz3rd5TCD0Ju73+7I7/ylGtmlSb/SBw/uWOZb/0SC0sevn29hPHYzFyG4aKCBo8mSwKhwr88jL4t//Pr355+uTvR6/vX2/uH4fKfP5/evn6xt1+/6G+vDydm/XR++/Xi0+vdv6LXk1/uPn/+zf49G0+Sz5+hH0q9SCbSIrPEqe96ILwgCcu8ev7rpSlfmsFl+QJeGcvSmOmYJkw/xeo5hcbBRli2kTLNkss97PQ8OAlOQpHG/AWQA0/6x8eiMxOpjO/01UJbawbAlBspUhsWP3SEyNKETk+CUTAKpZhtxfnYzRyCAcxDnq6EVmkClmCO72uPoq9PJ5wB2+Wx/1YsqMre0G8rzfF9r5gUsbPE95BgQ3aEHJpLZsWK00TFueRUc6Pkiv85ebpIG3KxSGnM1H379CKYBMMwFsaG/kUbCX00L31EDjM9DcYgaptxsscVxvC4mp4Fw1Fw5scJriqAV/sUvlXzuQkfjUrBeS15wmis2dzSk0ngXh7eC87T7Dw1AhS94KgiNiDQxI7XaDIZu5YcWxc11HQCWu3gMoaDq0eMyT2AAWs+HTmJ6AwiRXARpirmoAMTzrR6Bo6wYGrQz5hkacSdM7TREulmFzgdQlftZjDTDGIQf8nc+NDJaiCBE/4M20nz+RxfUodqoWBq0EOElEZYdLo2mOkY5GyfLAjp8gnjcO0g4KRDd5EUNMq1Qb3ZFoRJAiCTiRTCNDqoGgzkOgM2tzLbGVXKUSpo93YxCf3vze+VVJqCo13h9ryD85PZbhIFzmVEvWQO5K1r0sGURsxSl2VgTBUIvHW3TKnLLvAwfKilbsianeh1ZtW79OCZGvQxn+ULhN23e4c1Cg2EwjY9xHzOcomqoYJ4ixm3s0ASrQVqdBUEVoJzxj6ylK8aZFxrsATUt1cQEKkrxnFIpjMOXht6WBRxADOaNrgf8GkHu8sHUD4HgHx1OK5G6+JoJlqJqM9de+kKGIQzWCGdmWcJza2Q2KzWYWB8Xd6phK0EOFRQ+RNfH0BaR5fc3dJmmuMjdwBQJHjQsHjYo/g953qNcngEzOfEuU2XXpWvgkQ0kwtuNHdOGxeqxJTpwOZ5j8tqBt6xh6wCbdiqF3t0/bNaTahbqk7p/rnJs+orQA71SwVTg37OjKUx5xnlv+dMIv00kD5KtnsYjyzyQ8tmslqpYo7Ne/dHiMXPBQBhVHpNIwa5KMbfgPpEcxj63/dY05jmmAcqEaV/bJUMKo1+mTagMiK3SgMgy+MeGkB4Dqg8H027RO9iRPM9AzKB5kxmzGJ5bg3lPXWH0vI0spCzMglePaYzyaInanGf0/mN72cY6pndd0MLDv7ZxgLLozcYqPS6BF5INcMooNmFFaib3K8tH9MMClM0ONVQPpi0R1CHYqi3KRHT4cg71lYSVznMc0lRU6mhwAeOwEhqbxqMS2YoWCeW+FQQ4OrSslikYJgIRwHwDGftDH5vi86h4MestA6Dtd1VPIok1/AXJMfJNijIbX2WvH3TZEznUiyWmB1UEG/WbpTF4x7RkmuBxosK4he6S92LxyaRoUxrthb4GLcov9vSnr8Dir/YnrWyBXl/38nUT+M5Jp3VBCAg7+Fgs24rCKeqARG39WjomiVY7CwRLmZOOsTyoS9ze9fO90K31CfO2Ex2fYJsI9R3bQ5Id9rgMNHoGJrhmz4Lu1R5EdzxHOCg70u/3ta/5CvMo7vmUvquFFcqFhdZiMsXMK4doN+/azcOqaDA5bQnPtZQSFIhVYz7nAIAxj88D0YwSv/YShIUSQOqrh0gkHbFDakW1KyTmUIDUA2FDDERiYjoHJOrgiB5IVQKom/rbYPx0kzCzXMbF5QhfVQAAdMcdRShyVMsNJZSFgAwz9MuBr9n/R45XMHUpMfmLilKw3afkuTWL1fOEoxii/JbPRehe9OgSpnNNWRykUpcyoPQNZBludgmnkojjMg1l1+7X/c+5lage2ElAsnNlGbY95q59drh0dS7JO2qGd0zKsFgMZMuEUDc5dgA4Z0WTuIhyHrPIK5gIcG3gxSjru99Zkx7F8cODgkhRTjtOQ/ZgpCstcfjHzx5jqeFGrIl8DiYedRQ2AQ4lMvjZkbJHJ/OBhRTIx7oXTOSomVPC7dVPccYSgiyWZ9pLvOYUzQqbUHeeXdnBplWETfvshxLqkYH4Bljie/LHtxDxbXXhdsqxaJQiQCtdi248swUiiSFufs6DFkkALMwnP7jmV0gMutaJJphllMA/LHMOCweGhQG6kkILbB0MA9ZQ/nl1b4XVl2LOODmxHR45mm6joo2Nyw45MWonTSQ02H3UuuN2wfbXMG0Ty8y6k4iUXEr0PSs01IK0Aw1ug0G8XcFxmfukDv03e5oQXvb61Bmnvk7NNQfkGG0O8DpeedqsxZzowfPDdA0idcZLKYlj7CD1S3Il03dJwO5FqAkhKkA+BpiWG3qn4bwNmBSNsmswGrpgwfteBrUz5G/xYWwlwgktD1rlmXYrBSAsggvHvYoBBpofTtyhPcSL+iMGQ4VA8JSQzU9cEn1R8V5yGWKsQ+Pu3ey3nb6PAw+1Qd2qDD1Y/5dMY6/GTLu3Kx66xqGsu0tGm7TTJuOjzuRf4N26wfQuwK8VSOf/AIoV+MxorQcd+5K9OcPOofBuOnODpeudsjZsMODLo6d4xfH/s+D2z0sber9uCPSUx/Xdo9ID5VkWxc3lsNBl4GK+G9UriN+vBZal/auGMecbIzbne+H6n/37yosbui6C7kfiivE5Ms9ufP1KrlfisRcVheIE5YZkim5ngspyVxpUpS1hpSLzZByz5SI2hflqMj3EHIJf4l4ZsnPS8jPObn49NcfGjeFmVmnETE6uu6e6OoepXHiuUOO0Ta677RVR/c3zWHuXEsui/gbr5hS8h9BEFLls3fl1WLiqoq268k/FbSEhCH5twuiMHKRluohEWQIZMk1/0hAZ8DLiOFQyzPAuV1crzV4OYBRDwizkPvNoMwlKiV2ySuWQuTAd+MuEynJA6kW3zeF/OGnnWm9Covr4VdhcW39f9yoElbOLgAA) only loads standardjs-markdown + a hole bunch of other things ending up with 3.8mb download
![image](https://user-images.githubusercontent.com/1148376/131311115-433f43af-d7f8-417c-93e4-79290ac6eae4.png)

right now i'm very thankful for [npmgraph](https://npmgraph.js.org/?q=standard-markdown@7.0.0) existences

**Comment by ljharb** [comment]
The number of deps has no relation to “less JIT compiling” (nor is that something you can reliably ever optimize for).

Surfacing this kind of information defeats the point of dependencies imo - which is to serve as an abstraction. It shouldn’t matter whether one dep is large/complex and has zero deps of its own, or whether it’s got hundreds of deps composed in it (but is smaller and less complex in aggregate), except to encourage more dependency use, not less.

**Comment by MylesBorins** [comment, marked as answer]
hey @jimmywarting thanks for the suggestion.

I could see how have an idea of the full size of the tree would be useful in certain cases... you want to know not just direct dependencies but also transitive dependencies.

My concern with the proposed mock is that unless someone know exactly what that means it would be a confusing number. Fitting "dependencies" and "transitive dependencies" into the same line would also be a bit confusing.

To make matters even more confusing / fun... depending on your tree and if transitive dependencies have updated / changed at all the number of transitive dependencies is VERY dynamic and could change at any point... consistently calculating this for all packages on all page views to ensure that it is accurate seems... expensive.

Due simply to the dynamic nature of transitive deps I don't think that this is something that should be surfaced by default on the landing pages of packages.

Thoughts?

**Reply by jimmywarting** [reply]
hmm, get your point about dynamic numbers, seems like a both pro & con

> My concern with the proposed mock is that unless someone know exactly what that means it would be a confusing number. Fitting "dependencies" and "transitive dependencies" into the same line would also be a bit confusing.

it was just a quick inline edit on the webpages devtool (just to demonstrate what i wanted to have) how it would actually present this was never really the issue.

**Reply by MylesBorins** [reply]
@jimmywarting keeping things up to do from the serverside would be extremely burdensome / expensive / error-prone. For now I don't think the team will be exploring this request, but thank you for the suggestion.
